conda_libmamba_solver/solver.py

```diff
@@ -611,39 +611,31 @@ def _maybe_raise_for_problems(
         if hasattr(self.solver, "explain_problems"):
-            try:
-                raw_problems = self.solver.explain_problems()
-                key = None
-                sections = {}
-                for line in raw_problems.splitlines():
-                    if line.strip().startswith("===="):
-                        title = line.strip("=").strip()
-                        if "(old)" in title:
-                            key = "old"
-                        elif "(new)" in title:
-                            key = "new"
-                        else:
-                            key = "disclaimer"
-                        sections[key] = []
-                        continue
-                    if key is not None:
-                        sections[key].append(line)
-                return "\n".join(
-                    [
-                        *sections["old"],
-                        "",
-                        "***************************************************",
-                        " conda has set experimental_sat_error_message=true",
-                        " You will now see an explanation of the conflicts.",
-                        " You can provide feedback to the libmamba team at:",
-                        "  https://github.com/mamba-org/mamba/issues/2078",
-                        "***************************************************",
-                        "",
-                        *sections["new"],
-                    ]
-                )
-            except Exception as exc:
-                log.debug("Could not parse 'explain_problems'", exc_info=exc)
+            explained_problems = self.solver.explain_problems()
+            if "====" in explained_problems:
+                # This is for libmamba 1.3.x, where the explanation had a disclaimer
+                try:
+                    key = None
+                    sections = {}
+                    for line in explained_problems.splitlines():
+                        if line.strip().startswith("===="):
+                            title = line.strip("=").strip()
+                            if "(old)" in title:
+                                key = "old"
+                            elif "(new)" in title:
+                                key = "new"
+                            else:
+                                key = "disclaimer"
+                            sections[key] = []
+                            continue
+                        if key is not None:
+                            sections[key].append(line)
+                    return "\n".join(sections["old"] + sections["new"])
+                except Exception as exc:
+                    log.debug("Could not parse experimental 'explain_problems'", exc_info=exc)
+            else:
+                # libmamba 1.4.x enabled this by default, no need to parse disclaimers
+                return "\n".join([self.solver.problems_to_str(), explained_problems])
```

Comment:
Is there a chance that `problems_to_str()` fails and raises an exception?

Reply:
We have been calling it raw (uncaught) since the first release and didn't encounter any issues (e.g. https://github.com/conda/conda-libmamba-solver/blob/f7b559f422e9fb416a9c33b5d134fecd15f4262e/conda_libmamba_solver/solver.py#L580) like that, so I'd say the risk is minimum.